src/branch_view.js

```diff
@@ -2,19 +2,42 @@
 goog.provide('branch_view');
 goog.require('utils');
 goog.require('banner_css');
+goog.require('safejson');
 
 function renderHtmlBlob(parent, html) {
-	var re = /<script type="text\/javascript">((.|\s)*)<\/script>/;
-	var match = html.match(re);
+	var re, match;
+
+	// journey metadata
+	re = /<script type="application\/json">((.|\s)*?)<\/script>/;
+	match = html.match(re);
 	if(match) {
 		var src = match[1];
-		html.replace(re,'');
+		html = html.replace(re,'');
+
+		var metadata = safejson.parse(src);
+		if (metadata && metadata.injectorSelector) {
+			var parentTrap = document.querySelector(metadata.injectorSelector);
+			if (parentTrap) {
+				parent = parentTrap;
```

Comment:
It should be ok, because if there is no parent, it will default to document.body (on line 41).  The old branch.banner() used document.body at all times, so parent should be absent in most cases.